personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+                    table.pop(i)
+                else:
+                    i += 1
+            final_report = sorted(table, key = lambda x : x[sort_by_column], reverse = descend)
+    
+    # Generates a report showcasing required transactions.
+    elif report_meta.get("type", "M") == "T":
+        table = [[transaction["txn_date"], transaction["txn_detail"], \
+           transaction["debit"], transaction["credit"]] \
+           for transaction in transactions_all.values()]
+        if not sort_code.get("sort_req", False):
+            final_report = table
+        else:
+            sort_by_column = 2 if sort_code.get("sort_by", "D") == "D" else 3 if sort_code.get("sort_by", "D") == "C" else 0
+            descend = True if sort_code.get("order", "D") == "D" else False
+            i = 0
+            while i < len(table):
+                if table[i][sort_by_column] == 0:
+                    table.pop(i)
+                else:
+                    i += 1
+            final_report = sorted(table, key = lambda x : x[sort_by_column], reverse = descend)
+
+    # Generates a report showcasing key transactions of required months.
+    else:
+        for month_key in monthly_summary:
+            table = [[None, transactions_all[txn_key]["txn_date"], transactions_all[txn_key]["txn_detail"], \
+               transactions_all[txn_key]["debit"], transactions_all[txn_key]["credit"]] \
+               for txn_key in monthly_summary[month_key]["txn_keys"]]
+
+            sort_by_column = 3 if sort_code.get("sort_by", "D") == "D" else 4 if sort_code.get("sort_by", "D") == "C" else 1
+            descend = True if sort_code.get("order", "D") == "D" else False
+            i = 0
+            while i < len(table):
+                if table[i][sort_by_column] == 0:
+                    table.pop(i)
+                else:
+                    i += 1
+            temp_report = sorted(table, key = lambda x : x[sort_by_column], reverse = descend)
+            final_report.append([monthly_summary[month_key]["month_name"]])
+            #print(tabulate([], headers=monthly_summary[month_key]["month_name"], tablefmt="grid"))
+            final_report += temp_report[:(report_meta.get("txn_per_month", 3))]
+
+    return final_report
+
+
+def show_output(report, report_meta, view_count):
+    """
+    Displays the summary of transactions in a tabulated format.
+    """
+    if report_meta.get("type", "M") == "M":
+        headers = ["Month", "Debit", "Credit", "Surplus/Deficit"]
+    elif report_meta.get("type", "M") == "T":
+        headers = ["Transaction_Date", "Transaction_Details", "Debit", "Credit"]
+    else:
+        headers = ["Month", "Transaction_Date", "Transaction_Details", "Debit", "Credit"]
+        if view_count is None:
+            view_count = len(report)
+        else:
+            cnt = 0
+            for i in range(len(report)):
+                if report[i][0]:
+                    cnt += 1
+                if cnt > view_count:
```

Comment:
4 levels of nesting here. It can be rewritten in a better way.

else > else > for > if